pkgs/by-name/ke/key-rack/package.nix

```
@@ -0,0 +1,62 @@
+{
+  lib,
+  fetchFromGitLab,
+  stdenv,
+  rustPlatform,
+  cargo,
+  rustc,
+  libadwaita,
+  meson,
+  ninja,
+  nix-update-script,
+  pkg-config,
+  python3,
+  wrapGAppsHook4,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "key-rack";
+  version = "0-unstable-2024-06-21";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.gnome.org";
+    owner = "sophie-h";
+    repo = "key-rack";
+    rev = "d6081b814ee2f91bc3d49734aeebcf7f512ac19a";
+    hash = "sha256-apAvethMUpsosKxk2DT310AIjsF755u36xm3cMYpRaM=";
+  };
+
+  postPatch = ''
+    patchShebangs {,.}*
+  '';
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit (finalAttrs) src;
```

```suggestion
    inherit (finalAttrs) pname version src;
```

Just learned that, looks not bad